src/lib/UtapiClient.js

```diff
@@ -554,163 +623,94 @@ export default class UtapiClient {
-            cmds.push(
-                ['zremrangebyscore',
-                    generateStateKey(params, 'storageUtilized'),
-                    timestamp, timestamp],
-                ['zadd', generateStateKey(params, 'storageUtilized'),
-                    timestamp, actionCounter]);
+            let storageIndex;
+            let objectsIndex;
+            const cmdsLen = cmds.length;
+            const paramsArrLen = paramsArr.length;
+            const noErr = paramsArr.every((p, i) => {
+                // storage utilized counter
+                storageIndex = (i * (cmdsLen / paramsArrLen));
+                actionErr = results[storageIndex][0];
+                actionCounter = results[storageIndex][1];
+                if (actionErr) {
+                    log.error('error incrementing counter for push metric', {
+                        method: 'UtapiClient._genericPushMetricPutObject',
+                        metric: 'storage utilized',
+                        error: actionErr,
+                    });
+                    callback(errors.InternalError);
+                    return false;
+                }
+                cmds2.push(
+                    ['zremrangebyscore',
+                        generateStateKey(p, 'storageUtilized'),
+                        timestamp, timestamp],
+                    ['zadd', generateStateKey(p, 'storageUtilized'),
+                        timestamp, actionCounter]);
 
-            // number of objects counter
-            actionErr = results[1][0];
-            actionCounter = results[1][1];
-            if (actionErr) {
-                log.error('error incrementing counter for push metric', {
-                    method: 'UtapiClient._pushMetricCopyObject',
-                    metric: 'number of objects',
-                    error: actionErr,
-                });
-                return callback(errors.InternalError);
+                // number of objects counter
+                objectsIndex = (i * (cmdsLen / paramsArrLen)) + 1;
+                actionErr = results[objectsIndex][0];
+                actionCounter = results[objectsIndex][1];
+                if (actionErr) {
+                    log.error('error incrementing counter for push metric', {
+                        method: 'UtapiClient._genericPushMetricPutObject',
+                        metric: 'number of objects',
+                        error: actionErr,
+                    });
+                    callback(errors.InternalError);
+                    return false;
+                }
+                cmds2.push(
+                    ['zremrangebyscore',
+                        generateStateKey(p, 'numberOfObjects'),
+                        timestamp, timestamp],
+                    ['zadd', generateStateKey(p, 'numberOfObjects'),
+                        timestamp, actionCounter]);
+                return true;
+            });
+            if (noErr) {
+                return this.ds.batch(cmds2, callback);
```

Comment:
This step is unavoidable. We have to increment counters, get the results and then use the values to update the sorted sets with the value and timestamp.